Exiled.CustomItems/Events/PlayerHandler.cs

```diff
@@ -0,0 +1,55 @@
+// -----------------------------------------------------------------------
+// <copyright file="PlayerHandler.cs" company="Exiled Team">
+// Copyright (c) Exiled Team. All rights reserved.
+// Licensed under the CC BY-SA 3.0 license.
+// </copyright>
+// -----------------------------------------------------------------------
+
+namespace Exiled.CustomItems
+{
+    using Exiled.API.Extensions;
+    using Exiled.API.Features;
+    using Exiled.CustomItems.API.Features;
+    using Exiled.Events.EventArgs;
+
+    /// <summary>
+    /// Handles Player events for the CustomItem API.
+    /// </summary>
+    internal sealed class PlayerHandler
+    {
+        public void OnChangingRole(ChangingRoleEventArgs ev)
+        {
+            switch (ev.NewRole == RoleType.Spectator)
+            {
+                case true:
+                {
+                    foreach (Player player in Player.List)
+                    {
+                        if (player == ev.Player)
+                            continue;
+
+                        if (CustomItem.TryGet(player, out CustomItem item))
```

Comment:
You should only check if the player has this CustomItem in their hand and not any possible CustomItem.
That check should also be done on "hacky" custom items only.

Reply:
TryGet(this Player) has 2 overloads, one returns an IEnumerable<CustomItem> that checks for any item in their inventory, and one that returns a single CustomItem if it's in their hand.

Reply:
Yes, I know, but then you've to check if that item correspond to the right one (the `this` CustomItem) or another random CustomItem he's using.

Reply:
This event is not inside the CustomItem class, otherwise i'd just
`
if (Check(player.CurrentItem))
`

Reply:
Ah sorry, my bad, I thought it was there.